plugins/catalog-backend-module-github/src/providers/GithubEntityProvider.test.ts

```diff
@@ -716,4 +717,390 @@ describe('GithubEntityProvider', () => {
+          organization: 'test-org',
+          topics: [],
+        },
+        created: true,
+        deleted: false,
+        forced: false,
+        commits: [
+          {
+            added: ['new-file.yaml'],
+            removed: [],
+            modified: [],
+          },
+          {
+            added: ['catalog-info.yaml'],
+            removed: [],
+            modified: [],
+          },
+        ],
+      },
+    };
+    const url =
+      'https://github.com/test-org/test-repo/blob/main/catalog-info.yaml';
+    const expectedEntities = [
+      {
+        entity: {
+          apiVersion: 'backstage.io/v1alpha1',
+          kind: 'Location',
+          metadata: {
+            annotations: {
+              'backstage.io/managed-by-location': `url:${url}`,
+              'backstage.io/managed-by-origin-location': `url:${url}`,
+            },
+            name: 'generated-8688630f57e421bc85f12b9828ed7dad6aff3bb3',
+          },
+          spec: {
+            presence: 'optional',
+            target: `${url}`,
+            type: 'url',
+          },
+        },
+        locationKey: 'github-provider:default',
+      },
+    ];
+
+    await provider.onEvent(event);
+
+    expect(entityProviderConnection.applyMutation).toHaveBeenCalledTimes(1);
+    expect(entityProviderConnection.applyMutation).toHaveBeenCalledWith({
+      type: 'delta',
+      added: expectedEntities,
+      removed: [],
+    });
+  });
+
+  it('apply delta update on removed files from push event', async () => {
+    const schedule = new PersistingTaskRunner();
+    const config = new ConfigReader({
+      catalog: {
+        providers: {
+          github: {
+            organization: 'test-org',
+          },
+        },
+      },
+    });
+
+    const provider = GithubEntityProvider.fromConfig(config, {
+      logger,
+      schedule,
+    })[0];
+
+    const entityProviderConnection: EntityProviderConnection = {
+      applyMutation: jest.fn(),
+      refresh: jest.fn(),
+    };
+    await provider.connect(entityProviderConnection);
+
+    const event: EventParams = {
+      topic: 'github.push',
+      metadata: {
+        'x-github-event': 'push',
+      },
+      eventPayload: {
+        ref: 'refs/heads/main',
+        repository: {
+          name: 'teste-1',
+          url: 'https://github.com/test-org/test-repo',
+          default_branch: 'main',
+          stargazers: 0,
+          master_branch: 'main',
+          organization: 'test-org',
+          topics: [],
+        },
+        created: true,
+        deleted: false,
+        forced: false,
+        commits: [
+          {
+            added: ['new-file.yaml'],
+            removed: [],
+            modified: [],
+          },
+          {
+            added: [],
+            removed: ['catalog-info.yaml'],
+            modified: [],
+          },
+        ],
+      },
+    };
+    const url =
+      'https://github.com/test-org/test-repo/blob/main/catalog-info.yaml';
```

Comment:
```suggestion
      'https://github.com/test-org/test-repo/tree/main/catalog-info.yaml';
```

Reply:
there are a few similar cases

Reply:
I am a little confused about that now. 
The only place I use `tree` is in the keys from refresh process (here https://github.com/backstage/backstage/pull/14758/files#diff-cfcfe2edc138e4fd10a566578eb1a7c0bf859511b54fd5ca12b98564849fdd5aR348).

The other places are all using blob what make senses if you think this is always a file, not a directory. 

Should I change the refresh key for use `blob` too?

Maybe @kissmikijr could guide me

Reply:
Hmmm I do think blob is the right thing here for files. If you use the `tree` url in github for a file it will redirect to `blog`.

Reply:
Yea, I am not 100% sure when does the catalog uses tree/blob. For refresh keys it is for sure that in the [UrlReaderProcessor](https://github.com/backstage/backstage/blob/master/plugins/catalog-backend/src/modules/core/UrlReaderProcessor.ts#L97) when we persist the keys, the location target will become tree,  even if you used blob in your entity. So we will need to have it tree in the refresh_key lookup to have a match and trigger a refresh, but everywhere else I think we need to use blob

Reply:
I was, too 😄 

I check it at my installation of the `GithubEntityProvider` and saw that it uses `tree` there, too

At the entity loaded from the catalog-info.yaml file:
```yaml
backstage.io/managed-by-location: url:https://github.com/[org]/[repo]/tree/main/catalog-info.yaml
backstage.io/managed-by-origin-location: url:https://github.com/[org]/[repo]/blob/master/**/catalog-info.yaml
```

And at the Location entity:
```yaml
backstage.io/managed-by-location: url:https://github.com/[org]/[repo]/blob/master/**/catalog-info.yaml
```

At `GithubEntityProvider.createLocationUrl` it uses `blob`

```ts
return `${repository.url}/blob/${branch}/${catalogFile}`;
```
_(and in our case, we use `catalogPath: /**/catalog-info.yaml` at the config `catalogPath`)_
https://backstage.io/docs/integrations/github/discovery#configuration

I also found
```ts
      // GitHub is automatically redirecting tree urls to blob urls so it's
      // fine to pass a tree url.
```
at `getGithubFileFetchUrl`.

`UrlReaderProcessor.doRead` will execute `GithubUrlReader.search` for file paths containing `*`

This means there might be one Location pointing to many catalog-info.yaml files resulting in entities pointing to their file.

I would conclude that for simple cases, you may see `blob` at the refresh key and for cases with wildcard(s), you will/may see `tree` (at the simple case via recursive tree, it seems to have `blob` as well, at `readTree` not I guess).

Maybe, it makes sense to just add both variants to the refresh operation? 🤔 
What do you think @kissmikijr ?

Reply:
Nice investigation! I agree, we should just add both variant to be sure it gets refreshed.

Reply:
alternatively, we could normalize this to `blob` at `GithubUrlReader.search`

Reply:
Amazing investigation. I updated to use both keys for now, we can asks for @freben about normalize this at some point (to avoid this type of confusion)

Reply:
Sounds to me that we should be leaning into `blob`. The intent of the URLs is that they should represent what you see in the browser's location bar when looking at the file, and as mentioned above, if you try `tree` it'll redirect to `blob`.

Reply:
I agree `blob` make sense.

But if I change to only use `blob` the refresh isn't triggered. Maybe we need to update more references from `tree` to `blob` before that?

It's a little confuse why we are using `tree` over `blob` for files